src/plugins/embedurl.js

```diff
@@ -406,11 +406,21 @@ if (!CKEDITOR.plugins.get('embedurl')) {
 
 					// Sync dimensions and alignment with editor wrapper
 
+					let styles = null;
+
 					const stylesJSON = instance.element.getAttribute(
 						'data-styles'
 					);
 
-					let styles = stylesJSON ? JSON.parse(stylesJSON) : null;
+					if (stylesJSON) {
+						try {
+							styles = JSON.parse(stylesJSON);
+						} catch (e) {
+							console.warn(
+								'Unable to parse data-styles attribute!'
+							);
```

Comment:
Couple things here:

1. Linter may warn about unused variable `e` here, so either use it (in the log statement) or mark it as intentionally unused by starting it with an underscore (ie. `_error`). Note also that we try to avoid abbreviations if we can (ie. prefer `error` over `e`), except for things like `i` for loop variables.
2. We generally try to avoid anything that might resembele console spew in Liferay projects because it can be hard for customers to distinguish between bugs and mere diagnostic information (see [guidelines on this](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/debugging.md)); sometimes they see console output as a sign of bugginess and any "helpful" output we might include actually just confuses them. And they may be alarmed to see a message ending in an exclamation point. As such, I'd actually just swallow this error rather than log it — in some browsers, the stack trace that comes with `console.warn` looks particularly scary.